scanner/src/main/kotlin/scanners/scanoss/ScanOss.kt

```diff
@@ -47,30 +56,79 @@ class ScanOss internal constructor(
             ScanOss(scannerName, scannerConfig, downloaderConfig)
     }
 
+    private val config = ScanOssConfig.create(scannerConfig).also {
+        log.info { "The $scannerName API URL is ${it.apiUrl}." }
+    }
+
+    private val service = ScanOssService.create(config.apiUrl)
+
+    companion object {
+        // The name of the multipart body being sent. Has no importance.
+        private const val FAKE_WFP_FILE_NAME = "anon.wfp"
+    }
+
+    // TODO: Find out the best / cheapest way to query the SCANOSS server for its version.
     override val version = BuildConfig.SCANOSS_VERSION
+
     override val configuration = ""
 
+    /**
+     * The name of the file corresponding to the fingerprints can be sent to SCANOSS for more precise matches.
+     * However, for anonymity, a unique identifier should be generated and used instead. This property holds the
+     * mapping between the file paths and the unique identifiers. When receiving the response, the UUID will be
+     * replaced by the actual file path.
```

Comment:
Proposal:

> SCANOSS uses the file's path to increase confidence in snippet matches / to prioritize them. However, as exposing the file path from private repositories might be undesired, this map provides a way to anonymize paths before sending them to SCANOSS, and restore the original paths them afterwards.

@eeisegn, does that wording sound good to you, too?

Reply:
Should I put at least the extension after the UUID ?

Reply:
We should certainly do that, but I'd be fine with leaving it as a TODO for this PR and address it in a follow-up.